.github/workflows/rust.yml

```diff
@@ -22,10 +22,13 @@ jobs:
       with:
         toolchain: stable
 
+    - name: Install Nextest test runner
+      uses: taiki-e/install-action@nextest
```

Comment:
This move will certainly be impactful to reduce pipeline times. 

In addition, can you set up the caching of dependencies? It is generally more significant toward this aim. Check this link to see what I mean: https://stackoverflow.com/questions/64634535/github-actions-cache-rust-artifacts

Reply:
I was looking for a way to cache the artifacts to avoid rebuilding everything each time. This will further reduce the build time. Thank you for the link.

Reply:
The combination of **nextest** and **artifact caching** have significantly reduced the build and test time from over 4 minutes to just 43 seconds. Check it out.

![Screenshot from 2024-07-04 11-33-57](https://github.com/adorsys/didcomm-mediator-rs/assets/65020150/135747d8-f3fc-42a7-a635-91649767e624)

Reply:
Great 🎉

Can you further update the syntax of the workflow and bump up some action versions to avoid the following warnings?
 
![image](https://github.com/adorsys/didcomm-mediator-rs/assets/30264663/ef83be7c-b654-419b-9e1c-24171f7a394b)

Reply:
thanks for addressing that. Fixed now